src/components/MessageList.tsx

```diff
@@ -0,0 +1,448 @@
+import SplitView from '../split-view/SplitView';
+import SplitViewPane from '../split-view/SplitViewPane';
+import { nanoid } from '../../node_modules/nanoid';
+import { downloadFile } from '../helpers/downloadFile';
+
+export type Indicator =
+	| 'indicator-unvisible'
+	| 'indicator-edited'
+	| 'indicator-successful'
+	| 'indicator-unsuccessful';
+
+export interface ParsedMessageItem {
+	sessionId: string;
+	dictionary: string;
+	messageType: string;
+	message: object | string;
+	delay: number;
+}
+
+export interface ActMessageItem {
+	actBox: string;
+	fullServiceName: string;
+	methodName: string;
+	message: object | string;
+	delay: number;
+}
+
+export function isParsedMessageItem(object: unknown): object is ParsedMessageItem {
+	return (
+		typeof (object as ParsedMessageItem).sessionId === 'string'
+		&& typeof (object as ParsedMessageItem).messageType === 'string'
+		&& typeof (object as ParsedMessageItem).dictionary === 'string'
+		&& typeof (object as ParsedMessageItem).delay === 'number'
+		&& (typeof (object as ParsedMessageItem).message === 'string'
+			|| typeof (object as ParsedMessageItem).message === 'object')
+	);
+}
+
+export function isActMessageItem(object: unknown): object is ActMessageItem {
+	return (
+		typeof (object as ActMessageItem).actBox === 'string'
+		&& typeof (object as ActMessageItem).fullServiceName === 'string'
+		&& typeof (object as ActMessageItem).methodName === 'string'
+		&& typeof (object as ActMessageItem).delay === 'number'
+		&& (typeof (object as ActMessageItem).message === 'string'
+			|| typeof (object as ActMessageItem).message === 'object')
+	);
+}
+
+interface EditMessageProps {
+	editMessageMode: boolean;
+	editedMessageIndex: number;
+	indicators: Indicator[];
+}
+
+const Messages = () => {
+	const store = useStore();
+	const [switchValue, setSwitchValue] = useState(false);
+
+	const loadFromFile = (file: FileList | null) => {
+		if (file != null) {
+			const reader = new FileReader();
+			reader.readAsText(file.item(0) as Blob);
+			reader.onload = () => {
+				if (typeof reader.result === 'string') {
+					jsonMessagesFromString(reader.result);
+				}
+			};
+		}
+	};
+
+	const jsonMessagesFromString = (rawFromFile: string) => {
+		try {
+			const json = JSON.parse(rawFromFile);
+			store.clearParsedMessages();
+
+			for (let i = 0; i < json.length; i++) {
+				store.addParsedMessage(json[i]);
+			}
+		} catch (error) {
+			// eslint-disable-next-line no-alert
+			alert('Failed to read the file. Please, try to select another file');
+		}
+	};
+
+	useEffect(() => {
+		if (switchValue) {
+			store.setEditMessageMode(false);
+			store.setReplayMode(true);
+			replaySendMessage(store.getCurrentMessagesArray, 0);
+		} else {
+			store.setReplayMode(false);
+		}
+	}, [switchValue]);
+
+	const replaySendMessage = (array: ParsedMessageItem[] | ActMessageItem[], index: number) => {
+		if (store.isReplay && array.length > 0 && index < array.length) {
+			setTimeout(() => {
```

Comment:
I think messages should be sent sequentially. Awaiting before one message is sent. you should ask how this feature should be implemented

move store.isReplay to this component using useState.